bookmark.lua

```diff
@@ -32,76 +32,110 @@ jumpdrive.write_to_book = function(pos, sender)
 
 end
 
+local function hasNil(tPos)
+	if nil == tPos
+		or nil == tPos.x
+		or nil == tPos.y
+		or nil == tPos.z
+	then
+		return true
+	end
+	return false
+end
+
+local function sanitizeAndSetCoordinates(meta, tPos)
+	meta:set_int("x", jumpdrive.sanitize_coord(tPos.x))
+	meta:set_int("y", jumpdrive.sanitize_coord(tPos.y))
+	meta:set_int("z", jumpdrive.sanitize_coord(tPos.z))
+end
+
 jumpdrive.read_from_book = function(pos)
 	local meta = minetest.get_meta(pos)
 	local inv = meta:get_inventory()
 	local player_name = meta:get_string("owner")
 
-	if inv:contains_item("main", {name="default:book_written", count=1}) then
-		local stack = inv:remove_item("main", {name="default:book_written", count=1})
-		local stackMeta = stack:get_meta()
-
-		local text = stackMeta:get_string("text")
-		local data = minetest.deserialize(text)
-
-		if data == nil then
-			-- put book back, it may contain other information
-			inv:add_item("main", stack)
-			-- alert player
-			if nil ~= player_name then
-				minetest.chat_send_player(player_name, "Invalid data")
+	local iSize = inv:get_size("main")
+	local stack
+	local stackName
+	local stackMeta
```

Comment:
Main naming style seems to be snake case, while there's some places where this is mixed up already would be good to somewhat follow main coding style. that would mean naming it `inv_size`

Reply:
Also variables used inside deeper scopes (`stackMeta`, `stack`, `stackName`, `text`, `target_pos`) could be better also defined in deeper scopes, that would allow `luacheck` linter to catch and warn about attempts to use variable where it is not valid and through that prevent crashes caused by simple mistakes.